agent/php_mysqli.c

```diff
@@ -353,6 +353,31 @@ nr_status_t nr_php_mysqli_query_set_query(nr_php_object_handle_t handle,
   return NR_SUCCESS;
 }
 
+nr_status_t nr_php_mysqli_query_clear_query(nr_php_object_handle_t handle) {
+  zval* metadata = NULL;
+
+  /* If a metadata entry exists then clear the "query" tag from it.
+   *  If an entry does not exist then nothing needs to be done.
+   */
+  metadata = nr_php_mysqli_query_find(handle);
+  if (NULL == metadata) {
+    return NR_FAILURE;
```

Comment:
Is this an error condition (the comment above suggests it is not)? Does `nr_php_mysqli_query_clear_query` need to return any value? It's return value is not checked in `nr_php_prepared_prepare_general` where it's used.

Reply:
A good question - was just mirroring the API for the `set()` function.  In the current use case of this function it is not important to know if any data was actually cleared - just that if it exists it was cleared.  It is possible the `handle` has never been used before so there would be no data to clear.  So this particular case (lines 362-364) is not a failure but a consequence of a reasonable, possible state of the agent.

But the `NR_FAILURE` return code does give the caller the context that there was no existing data for the `handle` which might be useful in a future, different use?

Reply:
This PR can be merged with the code as is.